core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainRequest.java

```diff
@@ -245,6 +247,7 @@ public void readFrom(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        checkVersion(out.getVersion());
```

Comment:
The issue with doing this is that such a test will sometimes pass (having triggered the `IllegalStateException`) and sometimes it won't.  In particular, if the REST request is routed directly to the master node, there will be no exception.  The exception will only occur if the REST request is routed to the non-master node which is on a different version from master (and not backward compatible w.r.t. the explain API), in which case the exception will be triggered when routed to the master node.  

I did manually test these scenarios.  We can discuss options tomorrow.